ergotree-ir/src/mir/coll_append.rs

```diff
@@ -0,0 +1,124 @@
+use crate::serialization::op_code::OpCode;
+use crate::serialization::sigma_byte_reader::SigmaByteRead;
+use crate::serialization::sigma_byte_writer::SigmaByteWrite;
+use crate::serialization::SerializationError;
+use crate::serialization::SigmaSerializable;
+// use crate::types::stuple::STuple;
+use crate::types::stype::SType;
+
+
+use super::expr::Expr;
+use super::expr::InvalidArgumentError;
+use crate::has_opcode::HasStaticOpCode;
+
+/// Takes two collections as input and produces the concatenated collection (input + col2)
+#[derive(PartialEq, Eq, Debug, Clone)]
+pub struct Append {
+    /// Collection - First Parameter; first half of the combined collection
+    pub input: Box<Expr>,
+    /// Collection - Second Parameter; later half of the combined collection
+    pub col_2: Box<Expr>,
+}
+
+impl Append {
+    /// Create new object, returns an error if any of the requirements failed
+    pub fn new(input: Expr, col_2: Expr) -> Result<Self, InvalidArgumentError> {
+        let first_elem_type : SType = input.post_eval_tpe();
+        let second_elem_type : SType = col_2.post_eval_tpe();
+        match (first_elem_type, second_elem_type) {
+            (SType::SColl(x), SType::SColl(y)) => {
+                if x == y {
+                    Ok(Append{input: input.into(), col_2: col_2.into()})
+                } else {
+                    Err(InvalidArgumentError(format!(
+                        "Expected Append input and col_2 collection to have the same types; got input={0:?} col_2={1:?}",
+                        x, y)))
+                }
+            }
+            (SType::SColl(_), _) => {
+                Err(InvalidArgumentError(format!(
+                    "Expected Append col_2 param to be a collection; got col_2={:?}", col_2.tpe())))
+            }
+            (_, SType::SColl(_)) => {
+                Err(InvalidArgumentError(format!(
+                    "Expected Append input param to be a collection; got input={:?}", input.tpe())))   
+            },
+            (_, _) => {
+                Err(InvalidArgumentError(format!(
+                    "Expected Append input and col_2 param to be a collection; got input={:?} col_2={:?}", input.tpe(), col_2.tpe())))   
+            }
+        }
+    }
+
+    /// Type
+    pub fn tpe(&self) -> SType {
+        // Type is supposed to be the same on input and col_2
+        // Append::new checks types but later modifications are unchecked
+        // return type of first element
+        self.input.tpe()
+    }
+}
+
+impl HasStaticOpCode for Append {
+    // OpCode Append is already defined by previous programer; static number 67 assigned
```

Comment:
Seems like this comment does not provide any additional info.

Reply:
Can be deleted